substratevm/src/com.oracle.svm.test.jdk11/src/com/oracle/svm/test/jdk11/jfr/JFRTest.java

```diff
@@ -24,17 +24,65 @@
  */
 package com.oracle.svm.test.jdk11.jfr;
 
+import static org.junit.Assert.assertNotNull;
 import static org.junit.Assume.assumeTrue;
 
+import java.util.Set;
+
 import org.graalvm.nativeimage.ImageInfo;
+import org.junit.After;
+import org.junit.Assert;
+import org.junit.Before;
 import org.junit.BeforeClass;
 
 import com.oracle.svm.jfr.JfrEnabled;
+import com.oracle.svm.test.jdk11.jfr.utils.JFR;
+import com.oracle.svm.test.jdk11.jfr.utils.JFRFileParser;
+import com.oracle.svm.test.jdk11.jfr.utils.LocalJFR;
+
+import jdk.jfr.Recording;
+import jdk.jfr.consumer.RecordingFile;
 
 /** Base class for JFR unit tests. */
-public class JFRTest {
+public abstract class JFRTest {
+
+    protected JFR jfr;
+    protected Recording recording;
+
     @BeforeClass
     public static void checkForJFR() {
         assumeTrue("skipping JFR tests", !ImageInfo.inImageCode() || JfrEnabled.get());
     }
+
+    @Before
+    public void startRecording() {
+        try {
+            jfr = new LocalJFR();
+            recording = jfr.startRecording(getClass().getName());
+        } catch (Exception e) {
+            Assert.fail("Fail to start recording! Cause: " + e.getMessage());
+        }
+    }
+
+    /**
+     * @return List of all events (their names) that recording should contain.
+     */
+    public Set<String> searchForNextEvents() {
```

Comment:
I would remove this method for now as its result is not really used.

Reply:
Done.